app/lib/argocd_client.rb

```diff
@@ -17,4 +17,61 @@ def initialize
     @helm_chart_name = Figaro.env.HELM_CHART_NAME
     @helm_chart_version = Figaro.env.HELM_CHART_VERSION
     @helm_chart_repository = Figaro.env.HELM_CHART_REPOSITORY
+    @conn = Faraday.new(
+      url: @url,
+      headers: {
+          "Content-Type" => "application/json",
+          "Authorization": "Bearer " + @token
+      }
+    )
   end
+
+  def get_cluster_map()
+    puts("in get_cluster map")
+    cluster_map = JSON.parse(@conn.get('/api/v1/clusters').body)['items'].select { |i| i['connectionState']['status'] == 'Successful' }.map { |i| [i['name'], i['server']] }.to_h
+
+    cluster_map
+  end
+
+  def create_application(app_group_name, override_values)
+    output = @conn.post do | req |
+      req.body = {
+        metadata: {
+          name: "#{Figaro.env.argocd_project_name}-#{app_group_name}-#{Figaro.env.argocd_default_destination_name}",
```

Comment:
let's get `Figaro.env.argocd_default_destination_name` from the parameter instead

Reply:
done